### PR TITLE
Getting started with linear regression in R

### LINKED ISSUE #1799: Getting started with linear regression in R.
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Getting with started linear regression in R.
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction paragraph (2-3 paragraphs): 

Linear regression is a machine learning algorithm that is used to model a relationship between a continuous dependent variable and one or more independent variables. The independent variable can be either categorical or numerical. The case when we have only one independent variable then it is called as simple linear regression. If we have more than one independent variable, then it is called as multivariate regression.

In this tutorial, I will explain how to execute linear regression in R using some select functions and test its assumptions before we  can use it for a final prediction on test data.I will also explain how to obtain and understand the regression summary that help in determining the level of significance of the model.

### Key takeaways: 
(i) The leaner will understand how to perform linear regression using R.
(ii)  The leaner will be able to test the assumptions of the model before it  can be used for a final prediction on test data.
(iii) The learner will understand how to obtain and interpret linear regression statistics.
(iv) The leaner will be able to perform linear regression and draw important

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Madrinmarison**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1799

### CODE REVIEW

File: articles/linear-regression-r/index.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+This indicates that the newspaper is not significant and should be removed from our model. For a quick way for identification of which variable is significant to the model, we consider the asterisks(***). The three asterisks in the summary indicate highly significant coefficients to the model. The newsletter is not indicated hence not significant.
+
+(iv) F-statistic
+```r
+F-statistic: 570.3 on 3 and 196 DF,  p-value: < 2.2e-16
+
+```
+This statistic shows the overall significance of our model. Model is only significant if all coefficients are nonzero and it is not significant if all coefficients are zero. As we had already said in the coefficients section, coefficients being estimates they can never be zero. We there rely on p-value to declare whether our model is significant or not. A model is significant if its p-value less or equal to 0.05 otherwise it is insignificant. Our model has a p-value < 2.2e-16 which is much smaller hence our model is significant.
+
```
Comment (ahmadmardeni1) [inline articles/linear-regression-r/index.md:253]: Delete this part and explain every part of the code after writing it. For your reference:
```r
#Output
Call:
lm(formula = sales ~ youtube + facebook + newspaper, data = marketing)
```
It shows how `lm()` was called when it created the model.
.
.
And so on.

File: articles/linear-regression-r/index.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+Predplot<-Predplot+geom_line(aes(y=Predict, colour="Predicted"))
+Predplot<-Predplot + geom_point(aes(x=x_axis,y=Predict, colour="Predicted"))
+Predplot<-Predplot + geom_line(aes(y=test$sales, colour="Actual"))
+Predplot<-Predplot + scale_colour_manual("", values = c(Predicted = "red", Actual = "black"))
+Predplot
+
+```
+
+Running the above code we obtain the graph below.
+
```
Comment (ahmadmardeni1) [inline articles/linear-regression-r/index.md:274]: Please explain the plotting part. Beginners will not be familiar with it. Or make sure to add it as a prerequisite.

**Review by ahmadmardeni1 — COMMENTED** [review]
Good job! @Madrinmarison 🚀
Few changes, and it will be ready for the tech review.
Ping me when you are done.

**Comment by Madrinmarison** [comment]
@ahmadmardeni1 am done with the requested changes

**Comment by ahmadmardeni1** [comment]
Hello @Madrinmarison.🚀
Few changes and it will be ready for a final review.
I made some changes for your reference. Please look at them, then continue to edit the rest of your article.
Please start from the "Least Squares method" part to the end. Make sure you pull my changes first. Thank you.

**Comment by ahmadmardeni1** [comment]
Hello @Madrinmarison.🚀
Few changes, and it will be ready for a final review.
I made some changes for your reference. Please look at them, then continue to edit the rest of your article.
Start from the "Least Squares method" part to the end. Make sure you pull my changes first. Thank you.

**Review by ahmadmardeni1 — APPROVED** [review]
Hello @Madrinmarison.🚀
Few changes, and it will be ready for a final review.
I made some changes for your reference. Please look at them, then continue to edit the rest of your article.
Start from the "Least Squares method" part to the end.

**Review by ahmadmardeni1 — APPROVED** [review]

**Comment by ahmadmardeni1** [comment]
@Madrinmarison  No need for any change now. I fixed it. But please go through my changes 4124dbb and 147a58e. And make sure to prevent those mistakes in future articles.
Keep up the good work.👍 
Tech review in progress now.

File: articles/linear-regression-r/index.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+splitRation = 0.75
+set.seed(101) # to ensure same sample can be reproduced in future
+library(caTools) # Library for splitting data
+sample = sample.split(marketing$youtube, SplitRatio = splitRation)
+train = subset(marketing, sample==TRUE)
+TrainSize=dim(train)
+TestSize=dim(test)
+```
+
+![test-train_Size](image-one.png)
```
Comment (ahmadmardeni1) [inline articles/linear-regression-r/index.md:190]: This image does not exist. And the path is wrong. @Madrinmarison

**Review by ahmadmardeni1 — APPROVED** [review]

**Comment by Madrinmarison** [comment]
@ahmadmardeni1 I update the image.

**Comment by hectorkambow** [comment]
👋  Hello hello @Madrinmarison  

Hector here - i'll be doing the final review on your article(s). 

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on [plagiarism ](https://www.scribbr.com/category/plagiarism/)and what is considered as such.

[Getting started with linear regression in R #1942.pdf](https://github.com/section-engineering-education/engineering-education/files/6345459/Getting.started.with.linear.regression.in.R.1942.pdf)

**Comment by Madrinmarison** [comment]
@hectorkambow  I am done making the changes.

**Comment by hectorkambow** [comment]
👍  @Madrinmarison  great work 

Very minor edits during my final review for flow - please review when you get a moment. 🚀 
cc @ahmadmardeni1

**Pull request merged**